verifiable/onchain/credential.go

```diff
@@ -0,0 +1,61 @@
+package onchain
+
+type CredentialSubjectField struct {
+	Key      string `json:"key"`
+	Value    string `json:"value"`
+	RawValue string `json:"rawValue"` // hex representation of bytes
+}
+
+type State struct {
+	RootOfRoots        string `json:"rootOfRoots"`
+	ClaimsTreeRoot     string `json:"claimsTreeRoot"`
+	RevocationTreeRoot string `json:"revocationTreeRoot"`
+	Value              string `json:"value"`
+}
+
+type IssuerData struct {
+	ID    string `json:"id"`
+	State State  `json:"state"`
+}
+
+type CredentialSchema struct {
+	ID   string `json:"id"`
+	Type string `json:"_type"`
+}
+
+type MTP struct {
+	Root         string   `json:"root"`
+	Existence    bool     `json:"existence"`
+	Siblings     []string `json:"siblings"`
+	Index        string   `json:"index"`
+	Value        string   `json:"value"`
+	AuxExistence bool     `json:"auxExistence"`
+	AuxIndex     string   `json:"auxIndex"`
+	AuxValue     string   `json:"auxValue"`
+}
+
+type Proof struct {
+	Type       string     `json:"_type"`
+	CoreClaim  []string   `json:"coreClaim"`
+	IssuerData IssuerData `json:"issuerData"`
+	MTP        MTP        `json:"mtp"`
+}
+
+type CredentialStatus struct {
+	ID              string `json:"id"`
+	Type            string `json:"_type"`
+	RevocationNonce string `json:"revocationNonce"`
+}
+
+type Credential struct {
```

Comment:
is at a god idea to have such a structure public? Why we can't create some internal model and the W3C credential is the only credential that is public structure?

Reply:
I'm against this. Pretty sure it will confuse users of the library and ideologically I do not want to have a term onchain verifiable credential
@olomix @demonsh